experimental/Geometry/Misc.jl

```diff
@@ -0,0 +1,202 @@
+import AbstractAlgebra.Ring
+import AbstractAlgebra.RingElement
+import Oscar: radical_membership
+
+
+module Misc
+using Oscar
+
+export add_variables, divides_power
+
+####################################################################
+#  
+#  Miscellaneous routines used in the above 
+#
+####################################################################
+
+
+function add_variables( R::T, new_vars::Vector{String} ) where{ T<:MPolyRing } 
+  k = base_ring(R)
+  old_vars = String.( symbols(R) )
+  n = length( old_vars )
+  vars = vcat( old_vars, new_vars )
+  S, v = PolynomialRing( k, vars )
+  phi = AlgebraHomomorphism( R, S, gens(S)[1:n] )
+  y = v[n+1:length(v)]
+  return S, phi, y
+end
+
+function add_variables( P::T, new_vars::Vector{String} ) where{ T<:MPolyQuo } 
+  R = base_ring(P)
+  k = base_ring(R)
+  S, phi, y = add_variables( R, new_vars )
+  I = ideal( S, [ phi(g) for g in gens(modulus(P)) ] )
+  Q, projection = quo( S, I )
+  phi_quot = AlgebraHomomorphism( P, Q, [projection(g) for g in phi.image] )
+  return Q, phi_quot, [ projection(g) for g in y]
+end
+
+####################################################################
+#
+# Test whether some power of the polynomial f is contained in the 
+# principal ideal generated by g
+#
+function Oscar.radical_membership( f::U, g::U ) where{ U<:MPolyElem }
+  R = parent(f)
+  if R != parent(g) 
+    error( "Polynomials do not belong to the same ring" )
+  end
+  S, phi, t = add_variables( R, ["t"] )
+  I = ideal(S, [phi(g), one(S)-t[1]*phi(f)])
+  G = groebner_basis(I,complete_reduction=true)
+  if length(G)==1 && G[1] == one(S)
+    return true
+  end
+  return false
+end
+
+function Oscar.radical_membership( f::U, g::U ) where{ U<:MPolyQuoElem }
+  P = parent(f)
+  R = base_ring(P)
+  if P != parent(g) 
+    error( "Polynomials do not belong to the same ring" )
+  end
+  S, phi, t = add_variables( R, ["t"] )
+  I = ideal(S, 
+	    vcat( [phi(lift(g)), one(S)-t[1]*phi(lift(f))],
+		 [phi(h) for h in gens(modulus(P))] )
+	    )
+  G = groebner_basis(I,complete_reduction=true)
+  if length(G)==1 && G[1] == one(S)
+    return true
+  end
+  return false
+end
+
+##################################################################
+#
+# Checks whether some power of u is contained in the principal ideal 
+# generated by g and returns a solution (k,a) of the equation 
+#    u^k = a*g
+# If no such solution exists, it returns (-1,0).
+function divides_power( g::T, u::T ) where{ T<:MPolyElem }
+  println( "Check whether some power of $u is divisible by $g" )
+  R = parent(g)
+  parent(g) == parent(u) || error( "elements are not contained in the same ring" )
+  # Test for radical membership in the first place.
+  # TODO: Doesn't this computation already allow us to read 
+  # off our desired coefficients? How expensive is this compared 
+  # to what follows?
```

Comment:
It does indeed allow you read it off (this is e.g. described in "Ideals, varieties, and algorithms" by Cox,Little and O'Shea). Of course to do that, we can't do `one(R) in I` as I suggested above, you need the actual coefficients in the GB...

Reply:
exactly. And the question is what is the "Oscar"-way of doing that. See the discussion on Slack.